prov/sockets/src/sock_fabric.c

```diff
@@ -509,6 +514,38 @@ static int sock_ep_getinfo(const char *node, const char *service, uint64_t flags
 	return ret;
 }
 
+static void getnodename(char *buf, int buflen)
+{
+	int ret;
+	struct addrinfo ai, *rai = NULL;
+	struct ifaddrs *ifaddrs, *ifa;
+
+	gethostname(buf, buflen);
+	ret = getaddrinfo(buf, NULL, &ai, &rai);
+	if (!ret) {
+		freeaddrinfo(rai);
+		return;
+	}
+
+#if HAVE_GETIFADDRS
+	ret = getifaddrs(&ifaddrs);
+	for(ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
+		if (ifa->ifa_addr == NULL || (ifa->ifa_flags & IFF_LOOPBACK) || (ifa->ifa_addr->sa_family != AF_INET))
+			continue;
```

Comment:
We don't ignore the loopback interface, just give preferences for the real
ones. If don't find a real one, we return "127.0.0.1" as host/node name.

The problem is that the default behavior (i.e. NULL for source address)
fails for the "IP" provider.  For example, the utils/fi_info command
doesn't return "IP" at all.

Thanks,
    Lucho

On Tue, Feb 16, 2016 at 10:31 AM, Jithin Jose <notifications@github.com>
wrote:

> In prov/sockets/src/sock_fabric.c
> <https://github.com/ofiwg/libfabric/pull/1755#discussion_r53046038>:
>
> > +	int ret;
> > +	struct addrinfo ai, *rai = NULL;
> > +	struct ifaddrs *ifaddrs, *ifa;
> > +
> > +	gethostname(buf, buflen);
> > +	ret = getaddrinfo(buf, NULL, &ai, &rai);
> > +	if (!ret) {
> > +		freeaddrinfo(rai);
> > +		return;
> > +	}
> > +
> > +#if HAVE_GETIFADDRS
> > +	ret = getifaddrs(&ifaddrs);
> > +	for(ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
> > +		if (ifa->ifa_addr == NULL || (ifa->ifa_flags & IFF_LOOPBACK) || (ifa->ifa_addr->sa_family != AF_INET))
> > +			continue;
>
> Why do we want to ignore if it is a loopback address? If an application
> wants to use a particular source address, it should pass it through hints.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ofiwg/libfabric/pull/1755/files#r53046038>.
>